pallets/identity-management-mock/src/lib.rs

```diff
@@ -318,7 +321,12 @@ pub mod pallet {
 			let decrypted_identitty = Self::decrypt_with_tee_shielding_key(&encrypted_identity)?;
 			let identity = Identity::decode(&mut decrypted_identitty.as_slice())
 				.map_err(|_| Error::<T>::WrongDecodedType)?;
-
+			if let Identity::Substrate { network, .. } = identity {
+				ensure!(
+					network != SubstrateNetwork::Litentry,
+					Error::<T>::IdentityShouldBeDisallowed
+				);
+			}
```

Comment:
It's a bit off the mark, we actually allow the user to link another litentry account to his main litentry account. We need to check the network is Litentry and the address is the `user`